### PR TITLE
RUMM-2061 Disable RUM vitals monitor if no active foreground view

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt [hunk 1]
```diff
@@ -170,10 +171,11 @@ internal open class RumViewScope(
             else -> delegateEventToChildren(event, writer)
         }
 
-        return if (isViewComplete()) {
-            null
+        if (isViewComplete()) {
+            GlobalRum.updateRumContext(getRumContext().copy(viewType = RumViewType.NONE))
```
Comment (xgouchet) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt:175]: You should do that when stopping the current view only. The `isViewComplete()` can return true a long time after the view is stopped (e.g.: an activity goes in background, but there are ongoing network requests, so the view is still not complete). 
You could also introduce some issues if you navigate to another view, and 5 secondes later you update the rum context with your own...

File: dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumSessionScopeTest.kt [hunk 2]
```diff
@@ -647,6 +647,7 @@ internal class RumSessionScopeTest {
                 assertThat(it.cpuVitalMonitor).isInstanceOf(NoOpVitalMonitor::class.java)
                 assertThat(it.memoryVitalMonitor).isInstanceOf(NoOpVitalMonitor::class.java)
                 assertThat(it.frameRateVitalMonitor).isInstanceOf(NoOpVitalMonitor::class.java)
+                assertThat(it.type).isEqualTo(RumViewScope.RumViewType.BACKGROUND)
```
Comment (xgouchet) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumSessionScopeTest.kt:650]: You should also check that when a view is started by a `startView(…)` event, the type is `FOREGROUND`

File: dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScopeTest.kt [hunk 3]
```diff
@@ -295,6 +295,53 @@ internal class RumViewScopeTest {
...
+            mockViewUpdatePredicate,
+            type = fakeViewEventType
+        )
+
+        // Then
+        assertThat(GlobalRum.getRumContext().viewType).isEqualTo(fakeViewEventType)
+    }
+
+    @Test
+    fun `𝕄 update the context with viewType NONE W handleEvent(StopView)`(
```
Comment (xgouchet) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScopeTest.kt:326]: Can you add a test 
`"𝕄 not update the context with viewType NONE W handleEvent(StartView) {another view is}"` to ensure we don't mess up the context from the next view

**Review by xgouchet — CHANGES_REQUESTED** [review]

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt (hunk 1 above)
Comment (mariusc83) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScope.kt:175]: You're right, I am going to do the change here.

File: dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/vitals/VitalReaderRunnableTest.kt [hunk 4]
```diff
@@ -83,6 +94,28 @@ internal class VitalReaderRunnableTest {
...
+    @EnumSource(
+        value = RumViewScope.RumViewType::class,
+        names = ["FOREGROUND"],
+        mode = EnumSource.Mode.EXCLUDE
+    )
+    fun `𝕄 not read data, not notify observer but schedule 𝕎 run { viewType != FOREGROUND }()`(
+        viewType: RumViewScope.RumViewType
+    ) {
+        // Given
+        whenever(mockReader.readVitalData()) doReturn fakeValue
```
Comment (0xnm) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/vitals/VitalReaderRunnableTest.kt:107]: minor: this line is not needed

**Review by xgouchet — APPROVED** [review]

File: dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScopeTest.kt [hunk 5]
```diff
@@ -295,10 +295,152 @@ internal class RumViewScopeTest {
...
+            fakeEventTime,
+            fakeAttributes,
+            mockDetector,
+            mockCpuVitalMonitor,
+            mockMemoryVitalMonitor,
+            mockFrameRateVitalMonitor,
+            mockTimeProvider,
+            mockRumEventSourceProvider,
+            mockBuildSdkVersionProvider,
+            mockViewUpdatePredicate
```
Comment (0xnm) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/rum/internal/domain/scope/RumViewScopeTest.kt:393]: minor: we maybe need to write parameterized test here, by checking both `FOREGROUND` and `BACKGROUND` values, same for the test below.

**Review by 0xnm — APPROVED** [review]
lgtm! I added a comment about to improve the test, but it is non-blocking for me.

**Pull request merged**